core/node/api_server/src/execution_sandbox/mod.rs

```diff
@@ -417,3 +417,28 @@ impl BlockArgs {
         )
     }
 }
+
+/// VM executor capable of executing isolated transactions / calls (as opposed to batch execution).
+#[async_trait]
+trait OneshotExecutor<S: ReadStorage> {
+    type Tracers: Default;
+
+    async fn inspect_transaction(
+        &self,
+        storage: S,
+        env: OneshotEnv,
+        args: TxExecutionArgs,
+        tracers: Self::Tracers,
+    ) -> anyhow::Result<VmExecutionResultAndLogs>;
+
+    async fn inspect_transaction_with_bytecode_compression(
```

Comment:
I've recreated the logic previously used in the sandbox:

- Validation uses `inspect_transaction()` (which translates to `VmInterface::push_transaction` + `inspect()`)
- Execution uses `inspect_transaction_with_bytecode_compression` (which translates to `VmInterface::inspect_transaction_with_bytecode_compression`).

Not sure whether this can / should be simplified to a single method.

Reply:
I think it's fine to postpone this decision.